prov/opx/src/fi_opx_init.c

```diff
@@ -396,6 +396,9 @@ static int fi_opx_fillinfo(struct fi_info *fi, const char *node,
 			fi->ep_attr->rx_ctx_cnt = hints->ep_attr->rx_ctx_cnt;	/* TODO - check */
 	}
 
+	fi->nic = ofi_nic_dup(NULL);
+	fi->nic->bus_attr->bus_type = FI_BUS_PCI;
```

Comment:
Are the pci attributes set anywhere? otherwise may just leave the bus type as unknown.

Reply:
@lsavers Can you answer this question?

Reply:
In discussing with Lindsay, we needed to initialize the NIC in order to get (we believe) NCCL to work; but we aren't sure whether we really needed to set bus type or just call ofi_nic_dup(NULL). Lindsay is going to retest just without setting bus type today, but we think at least line 399 is necessary to get past the issue she was seeing.

Reply:
I was able to verify that both lines are needed to get past a NCCL initialization error for OFI_NCCL_PROTOCOL=RDMA (https://github.com/aws/aws-ofi-nccl/blob/ecbf29867f0f772aea2767447dd126ebdbc45546/src/nccl_ofi_topo.c#L123).  We do have a future work item to add code to set the additional attribute values; however, this work has not yet been completed.

Reply:
thanks for the verification.